priorityq.py

```diff
@@ -4,57 +4,103 @@
 
 class Element(object):
     ''' Create data element with default value and next pointer. '''
-    def __init__(self, value, priority=0, behind=None):
+    def __init__(self, value, priority=0):
         ''' Value and next pointer default to none. '''
         self.val = value
-        self.behind = behind
         self.prio = priority
 
 
 class Priorityq(object):
-    ''' Create an empty queue. '''
+    ''' Create an empty heap. '''
     def __init__(self):
-        self.front = None
-        self.back = None
+        self.binlist = []
 
     def insert(self, value, priority=0):
-        ''' Add data element to the back of queue. '''
-        new_element = Element(value, priority)
-        current = self.front
+        ''' Add a value to the heap. '''
+        self.binlist.append(Element(value, priority))
+        self._promote(len(self.binlist)-1)
+        return
 
+    def _parent(self, index):
+        ''' Find the parents of a bin. '''
+        return [(index-1)//2, self.binlist[(index-1)//2].prio]
```

Comment:
repeating the math to get the parent index twice feels janky to me.  I'd rather see you bind a local symbol to that value and then use the symbol:

```python
idx = (index - 1) // 2
return [idx, self.binlist[idx].prio
```